pkg/build/controller/factory/factory.go

```diff
@@ -26,8 +26,14 @@ import (
 )
 
 // logAndRetry retries forever - BuildPodController currently has no fatal errors
-func logAndRetry(obj interface{}, err error, _ int) bool {
+func logAndRetry(obj interface{}, err error, count int) bool {
 	kutil.HandleError(err)
+	if count > 60 {
```

Comment:
the comment should really be on the func, and then you can simplify this to return count < 60.  I would rename the method to be limitedLogAndRetry.